packages/core/src/shared/vscode/commands2.ts

```diff
@@ -520,7 +520,9 @@ function handleBadCompositeKey(data: { id: string; args: any[]; compositeKey: Co
     Object.entries(compositeKey).forEach(([index, field]) => {
         const indexAsInt = parseInt(index)
         const arg = args[indexAsInt]
-        if (field === 'source' && typeof arg !== 'string') {
+        if (field === 'source' && !arg) {
```

Comment:
this would allow `arg=0` or `arg=''`, which is semantically not what we want. we specifically want to allow `undefined`, because that particular case is likely just "absence of an arg". whereas `0`, `''`, etc are not "absence" but rather "an arg was actually passed but likely a mistake".